lib/utils.js

```diff
@@ -16,44 +19,55 @@ const specialProperties = ['__proto__', 'constructor', 'prototype'];
  */
 
 const clone = exports.clone = function clone(obj, options) {
-  if (obj === undefined || obj === null)
+  if (obj === null || primitiveTypes.has(typeof obj)) {
```

Comment:
Is the primitiveTypes.has() check necessary here? We can probably just do typeof obj !== 'object'